tests/stub/summary/test_summary.py

```diff
@@ -247,6 +235,793 @@ def test_multiple_notifications(self):
+            self.make_test_notification(severity="INFORMATION")
+        ]
+        summary = self._get_summary(
+            "summary_with_notifications_omitted_result.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            self.assert_is_omitted_result(summary.gql_status_objects[0])
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+        self.assert_is_test_notification_as_gql_status_object(
+            summary.gql_status_objects[1],
+            severity="INFORMATION", parsed_severity="INFORMATION"
+        )
+
+    def test_information_no_data(self):
+        in_notifications = [
+            self.make_test_notification(severity="INFORMATION")
+        ]
+        summary = self._get_summary(
+            "summary_with_notifications_no_data.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            self.assert_is_no_data(summary.gql_status_objects[0])
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+        self.assert_is_test_notification_as_gql_status_object(
+            summary.gql_status_objects[1],
+            severity="INFORMATION", parsed_severity="INFORMATION"
+        )
+
+    def test_unknown_severity(self):
+        in_notifications = [
+            self.make_test_notification(severity="FOOBAR")
+        ]
+        summary = self._get_summary(
+            "summary_with_notifications.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            self.assert_is_success(summary.gql_status_objects[0])
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+        self.assert_is_test_notification_as_gql_status_object(
+            summary.gql_status_objects[1],
+            severity="FOOBAR", parsed_severity="UNKNOWN"
```

Comment:
```suggestion
            severity="FOOBAR", parsed_severity=None
```
Java driver does not have the `UNKNOWN` parsed value, it returns an empty `Optional` is such cases. The backend can simulate `UNKNOWN` value for Testkit, but it lead to some other tests failure.

Reply:
That was me misreading the ADR. This change has been reverted after having a chat with @bigmontz as well.